### ISSUE TITLE
Push notifications. Hostname/IP doesn't match certificate's altnames

### ISSUE THREAD

**Issue by vytasmk** [RocketChat/Rocket.Chat#16161]
### Description:

Push notifications are not working. 

Was looking thru the Issues did not find any mention about such error.

I have upgraded our chat from 0.71.1 to the latest version 2.3.1.

### Steps to reproduce:

1. In Adminstration go to Push section
2. All settings are default using `https://gateway.rocket.chat` gateway
3. Use "Send a test push to my user" button to send a test message
4. Message is not sent and error is logged

### Server Setup Information:

- Version of Rocket.Chat Server: 2.3.1
- Operating System: Debian 9.11
- Deployment Method: Docker
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version: v8.15.1
- MongoDB Version: 4.2.2

### Client Setup Information
iPhone with official RocketChat App

### Additional context
Our Rocketchat service is running inside Docker container using official Docker container. Docker container binds to localhost port and from outside it is accessed thru our haproxy service which terminates SSL connection. 

Our sertificate is with wildcard (if that is important) generated by Let's Encrypt service. 

Certificate has CN: *.domain.com and two altnames: "*.domain.com, domain.com". 

Docker Environment settings:
```
ROOT_URL=http://localhost:3000
HTTPS_PROXY=https://chat.domain.com
MONGO_URL=mongodb://mongodb:27017/meteor
MONGO_OPLOG_URL=mongodb://mongodb:27017/local
```

Can somebody point me where is the problem as the old version was working fine (HAProxy config is the same)?

### Relevant logs:

```
rocketchat_1            | server.js:204 System ➔ error Error sending push to gateway (3 try) -> { Error: tunneling socket could not be established, cause=Hostname/IP doesn't match certificate's altnames: "Host: gateway.rocket.chat. is not in the cert's altnames: DNS:*.domain.com, DNS:domain.com"
rocketchat_1            |     at ClientRequest.onError (/app/bundle/programs/server/npm/node_modules/meteor/http/node_modules/tunnel-agent/index.js:177:17)
rocketchat_1            |     at Object.onceWrapper (events.js:315:30)
rocketchat_1            |     at emitOne (events.js:116:13)
rocketchat_1            |     at ClientRequest.emit (events.js:211:7)
rocketchat_1            |     at TLSSocket.socketErrorListener (_http_client.js:401:9)
rocketchat_1            |     at emitOne (events.js:116:13)
rocketchat_1            |     at TLSSocket.emit (events.js:211:7)
rocketchat_1            |     at emitErrorNT (internal/streams/destroy.js:66:8)
rocketchat_1            |     at _combinedTickCallback (internal/process/next_tick.js:139:11)
rocketchat_1            |     at process._tickDomainCallback (internal/process/next_tick.js:219:9) code: 'ECONNRESET' }
```

**Comment by geekgonecrazy** [comment]
Does it actually say DNS:*domain.com and DNS:domain.com or did you replace your company domain with that?

It looks like its not actually proxying... so when https://gateway.rocket.chat your proxy is responding.  Which would explain the cert mismatch related error

**Comment by vytasmk** [comment]
Hello. Yes I have changed my domain to "domain.com".

Regarding second idea, need to check so it means that our proxy is responding when Rocketchat service is trying to reach https://gateway.rocket.chat 

But why it was working with older version as I have just changed docket container and in proxy just changed port number. So it should not be a problem... I will check what I get from inside container if I try to reach the gateway address.

**Comment by geekgonecrazy** [comment]
> I have just changed docket container and in proxy just changed port number.

Which port number did you change that proxy or Rocket.Chats?

> I will check what I get from inside container if I try to reach the gateway address.

This should hopefully give some clues

**Comment by vytasmk** [comment]
Port inside container is default `3000`. And on host system is bind to `locahost:8032`. HAProxy is running on the host and is bind to the public IP. It proxies all requests to that port on localhost. But I think here is all OK as everything else is working just fine.

Regarding communication with gateway I have tried to connect to the gateway using simple curl tool to check the servers certificate.

```
rocketchat@rocket:/app/bundle$ curl --insecure -v https://gateway.rocket.chat 2>&1 | awk 'BEGIN { cert=0 } /^\* SSL connection/ { cert=1 } /^\*/ { if (cert) print }'
```

And the result is:
```
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* Server certificate:
* 	 subject: CN=*.rocket.chat
* 	 start date: 2019-12-21 00:00:00 GMT
* 	 expire date: 2022-01-17 23:59:59 GMT
* 	 issuer: C=GB; ST=Greater Manchester; L=Salford; O=Sectigo Limited; CN=Sectigo RSA Domain Validation Secure Server CA
* 	 SSL certificate verify ok.
* Connection #0 to host gateway.rocket.chat left intact
```

So it looks like I can reach the official gateway without problems. I still do not understand how it checks for gateway address in our proxy servers certificate...

**Comment by geekgonecrazy** [comment]
Interesting looks like some how Rocket.Chat might not be honoring that proxy config.

@RocketChat/backend any ideas?

**Comment by vytasmk** [comment]
Yes, so that is why I have submitted this issue, as there is no logic at least at this moment. It looks like it is trying to reach gateway but DNS resolves to our proxy server IP. But I have just created new Docker container with new version and old data. Old container I still have but it is now turned off. Proxy server config has changed just IP port on which new Docker container listens and nothing more and from inside the container I can reach gateway without problems so it looks like from inside rocketchat service it gets lost...

**Comment by vytasmk** [comment]
OK, I guess I found, it was mistake by me (not surprised) in configuration. 

As I mentioned I am using docker-compose and as example was using this docker-compose.yml (https://github.com/RocketChat/Rocket.Chat/blob/develop/docker-compose.yml)

 There was one environment parameter `HTTPS_PROXY=http://proxy.domain.com` that I have added to the new config as I was thinking that it is used to inform rocket chat that I am using proxy to reach rocketchat service. So it looks like I was wrong here. I have later removed it but it seems like I did not fully recreated Docker container. Today I was playing with it and fully deleted container and recreated it again - and it is working now!

So it would be great if somebody can update Docker Hub description to include what environment parameter are available and what they are really for.

Sorry for some noise but hope if somebody gets in similar problem will have some
